sdk/cpp/core/src/path/data_node.cpp

```diff
@@ -114,11 +114,25 @@ ydk::path::DataNodeImpl::populate_new_schemas_from_path(const std::string& path)
 ydk::path::DataNode&
 ydk::path::DataNodeImpl::create_datanode(const std::string& path, const std::string& value)
 {
+    std::string v = value;
```

Comment:
Suggest moving lines 117-129 into a separate function called something like '`escape_xml_declaration()`'. This is so that it is clear what is being done by these lines of code.